commit message test

Noamshalev1 opened the pull request.

Noamshalev1 committed:
commit message test

tomdugma commented on Project/src/main/java/com/example/demo/controller/PlayerController.java:
```diff
@@ -27,4 +27,5 @@ public String[] getPlayer(@PathVariable String id) {
     public List<String[]> getAllPlayers() {
         return service.getAllPlayers();
     }
+
```
comment test

tomdugma commented on Project/src/main/java/com/example/demo/controller/PlayerController.java:
```diff
@@ -27,4 +27,5 @@ public String[] getPlayer(@PathVariable String id) {
     public List<String[]> getAllPlayers() {
```
another comment

tomdugma reviewed (changes requested):
this is my comments

Noamshalev1 committed:
commit message test2

Noamshalev1 committed:
Exception Handler implementation

Noamshalev1 committed:
Update class to throw exceptions

tomdugma commented on Project/src/main/java/com/example/demo/service/Player.java:
```diff
@@ -33,11 +33,21 @@ public void loadPlayersData() {
 
     public List<String[]> getPlayers() {
         Collection<String[]> values = playerMap.values();
-        return new ArrayList<>(values);
```
Player is a model class, its bad practice to have business logic in here

the only thing Player should have is the fields of a player (id, age, ..) consructor + setters + getters (use lombok for that)

Please move the entire logic to a different class, like FileLoaderService

Noamshalev1 replied:
Done

tomdugma commented on Project/src/main/java/com/example/demo/exception/RestResponseEntityExceptionHandler.java:
```diff
@@ -0,0 +1,36 @@
+    protected ResponseEntity<Object> handleConflict(
+            RuntimeException ex, WebRequest request) {
+        String bodyOfResponse = "This should be application specific";
+        return handleExceptionInternal(ex, bodyOfResponse,
```
where is the implementation of handleExceptionInternal function ?

tomdugma commented on Project/src/main/java/com/example/demo/exception/RestResponseEntityExceptionHandler.java:
```diff
@@ -0,0 +1,36 @@
+        String bodyOfResponse = "Player not found: " + ex.getMessage();
+        return new ResponseEntity<>(bodyOfResponse, new HttpHeaders(), HttpStatus.NOT_FOUND);
+    }
+}
```
its good practice to have a "fallback" method for general exception
all exceptions extend Exception.class (the most generic java exception)

so if suddenly your app throws an exception you didnt think of, it will fall back to it

so make another method with 
@ExceptionHandler(Exception.class)

Noamshalev1 committed:
Update player to an object at model package
Create DeadPlayer which extend from Player

Noamshalev1 committed:
Update Interface IPlayerService
Update PlayerService to return Player

Noamshalev1 committed:
Create DTO to handle loading the data from csv and manage the hashmap

Noamshalev1 committed:
Update controller

Noamshalev1 committed:
Add new Exception
Update the general exception in RestResponseEntityExceptionHandler

Noamshalev1 committed:
Add new functions: save and update
Add @Autowired on the DAO (for the dependencies)

Noamshalev1 committed:
Update an annotation for ignoring the DataSourceAutoConfiguration

Noamshalev1 committed:
PlayerController: Create 2 new requests: POST and PUT
LoadDataService: Create 2 new functions: save and update

tomdugma commented on Project/src/main/java/com/example/demo/model/Player.java:
```diff
@@ -0,0 +1,176 @@
+
+/**
+ * This class holds details about Player (alive)
+ */
```
@AllArgsConstructor
@Getter
@Setter

tomdugma commented on Project/src/main/java/com/example/demo/service/LoadDataService.java:
```diff
@@ -0,0 +1,147 @@
+        if (playerMap.get(id) == null){
+            throw new PlayerNotFoundException("Player with id: " + id +" doesn't exist");
+        }
+        else{
```
no need else

tomdugma commented on Project/src/main/java/com/example/demo/service/LoadDataService.java:
```diff
@@ -0,0 +1,147 @@
+     * @param player (send as JSON format and cast to a player)
+     */
+    public void savePlayer(Player player){
+        if (player == null){
```
can be done via annotation

Noamshalev1 committed:
Add fallback for other exceptions
Add @Valid annotation for the object player
Remove unnecessary code
Add annotations using lombok
Next mission - tests
